app/models/metasploit/credential/pkcs12.rb

```diff
@@ -28,11 +37,52 @@ class Metasploit::Credential::Pkcs12 < Metasploit::Credential::Private
   # Method Validations
   #
 
+  validate :data_format
+
   validate :readable
 
+  #
+  # Class methods
+  #
+
+  # @param [Integer] pkcs12 The Base64-encoded  Pkcs12 certificate
+  # @param [String,nil] ca The CA that issued the certificate
+  # @param [String,nil] adcs_template The certificate template used to issue the certificate
+  # @return [String]
+  # @raise [ArgumentError] if an option is invalid
+  def self.build_data(pkcs12:, ca: nil, adcs_template: nil)
+    raise ArgumentError.new('pkcs12 must be set') if pkcs12.nil?
+    raise ArgumentError.new('ca must be a non-empty string') if ca && (!ca.is_a?(String) || ca.empty?)
+    raise ArgumentError.new('adcs_template must be a non-empty string') if adcs_template && (!adcs_template.is_a?(String) || adcs_template.empty?)
+
+    "msf_pkcs12:#{pkcs12}:#{ca}:#{adcs_template}"
+  end
+
   #
   # Instance Methods
   #
+  #
+
+  # The Base64-encoded Pkcs12 certificate
+  #
+  # @return [String]
+  def pkcs12
+    parsed_data[:pkcs12]
```

Comment:
Wouldn't we want something to de-base64 encode and then load the DER data into a certificate object we can use?

If that doesn't fit here because this is for presentation, then maybe at least adding another method to handle that conversion.

Reply:
I believe the `#openssl_pkcs12` method below does something similar. It returns a `OpenSSL::PKCS12` object, from which we can extract the certificate.

So we have `#pkcs12` returning the based-64 encoded Pkcs12 and `#openssl_pkcs12` returning a `OpenSSL::PKCS12` object.

I'm also happy to change this logic if needed.